scaleway/resource_baremetal_server.go

```diff
@@ -305,15 +377,30 @@ func resourceScalewayBaremetalServerRead(ctx context.Context, d *schema.Resource
 	_ = d.Set("description", server.Description)
 	_ = d.Set("options", flattenBaremetalOptions(server.Zone, server.Options))
 
+	listPrivateNetworks, err := baremetalPrivateNetworkAPI.ListServerPrivateNetworks(&baremetal.PrivateNetworkAPIListServerPrivateNetworksRequest{
+		Zone:     server.Zone,
+		ServerID: &server.ID,
+	})
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("failed to list server's private networks: %w", err))
+	}
+	_ = d.Set("private_network", flattenBaremetalPrivateNetworks(server.Zone, listPrivateNetworks.ServerPrivateNetworks))
```

Comment:
Why the zone is not returned on the `ServerPrivateNetworks`? This should be added

Reply:
It's returned in the id